### ISSUE TITLE
macOS notarization not yet possible

### ISSUE THREAD

**Issue by BoboTiG** [Legrandin/pycryptodome#381]
Hello,

I hope I am at the good place for that issue.

I am trying to make an application using PyCrpytodomex to be compliant on macOS (the famous "notarization"). Before Feb 2020, all went good. But now that Apple uses an enforced security, all applications must be signed and notarized.

I am hitting a "blocker" because it seems that compiled files of this project are using an SDK older than 10.9, and I got plenty of errors from the notarization report:
```json
{
    "severity": "error",
    "code": null,
    "path": "nuxeo-drive-4.4.2.dmg/Nuxeo Drive.app/Contents/MacOS/Cryptodome/Hash/_SHA256.cpython-37m-darwin.so",
    "message": "The binary uses an SDK older than the 10.9 SDK.",
    "docUrl": null,
    "architecture": "i386"
},
{
    "severity": "error",
    "code": null,
    "path": "nuxeo-drive-4.4.2.dmg/Nuxeo Drive.app/Contents/MacOS/Cryptodome/Hash/_SHA256.cpython-37m-darwin.so",
    "message": "The binary uses an SDK older than the 10.9 SDK.",
    "docUrl": null,
    "architecture": "x86_64"
},
```

According to Apple (https://developer.apple.com/documentation/xcode/notarizing_macos_software_before_distribution/resolving_common_notarization_issues#3087723), using 10.10 should be OK for retro-compatibilty too.

Do you are aware of that issue? And do you think you could do something about it?

Thanks for your attention :)

**Comment by Legrandin** [comment]
I was not aware of the issue. It is due to `cibuildwheel`, which is used to create MacOS binary wheels. Today, for anything older than Python 3.8, it creates "fat" binaries with both 32-bit and 64-bit code. The latest Xcode that supports 32-bit builds is version 9, so it's not possible to move to 10.

The next version of `cibuildwheel` will only build for 64-bit, so it will be possible to switch to 10.9 or later therefore fixing this issue. If it's causing major problems, I could rebuild the MacOS wheels with the version of `cibuildwheel` on master (without waiting for the official release)... I don't believe many people are using the 32-bit code anyway on MacOS.

**Comment by BoboTiG** [comment]
Thank you for the explanation.

If I may, this is a blocker issue for my company as Apple will block any non-notarized app. So I would love to see builds with 10.10 :)

BTW I am working with Python 3.7.

**Comment by colatkinson** [comment]
@Legrandin This isn't purely due to 32-bit binaries being produced. It appears to be a regression due to commit [8e699e7](https://github.com/Legrandin/pycryptodome/commit/8e699e76d645d60fc159af2a0a50580851a8e8ac). 3.9.4 (and I'm guessing 3.9.5) notarize successfully, whereas 3.9.6 does not.

Output of `otool -l` for a random 3.9.4 shared lib:

```
Load command 7
      cmd LC_VERSION_MIN_MACOSX
  cmdsize 16
  version 10.6
      sdk 10.13
```

And for the same lib in 3.9.6:

```
Load command 7
      cmd LC_VERSION_MIN_MACOSX
  cmdsize 16
  version 10.6
      sdk 9.4.1
```

For a couple of similar bugs:

* https://bugreports.qt.io/browse/PYSIDE-1066
* https://bugs.launchpad.net/lxml/+bug/1846788

In both cases, the build script is explicitly setting the SDK root using `xcrun --show-sdk-path` which, as it turns out, will actually return different paths nondeterministically. This can (apparently) be fixed via `xcrun --sdk macosx --show-sdk-path`, which seems to take a totally different code path that reliably gets the correct path. Alternatively, I believe that explicitly specifying the target SDK version will also work.

As an additional note, the last version of the SDK that supported 32-bit builds was actually [10.13](https://developer.apple.com/documentation/xcode_release_notes/xcode_10_release_notes?language=objc) (which was the SDK version used by XCode 9). Since the current CI config appears to be using XCode 9.4, the builds can *already* use an SDK version >= 10.0.

TL;DR This can be fixed while keeping 32-bit compat. The options to `xcrun` are just magic and full of (undocumented) nondeterminism.

**Comment by Legrandin** [comment]
Thanks @colatkinson, I will release a new version with your patch shortly.

**Comment by Legrandin** [comment]
@BoboTiG could you check if the new v3.9.7 (now on pypi) works for you?

**Comment by BoboTiG** [comment]
This works like a charm! :champagne: 
Thank you both for the very quick answer + fix :muscle:
